### PR TITLE
Readme fixes and additions

### PR DESCRIPTION
**Pull request by liorn**
Added more detailed instructions on how to connect a Jira environment, fixed documentation around rate limit defaults, and added the required Jira version 

---

_Additional context for reviewer_

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

**Comment by ori-moisis** [comment]
The change itself looks good
Since this PR should have no release notes or user impact, please clear those sections so they say "None", e.g
```
---
Release Notes:
_None_
```

**Comment by liorn** [comment]
@ori-moisis done. Thanks

**Review by ori-moisis — APPROVED** [review]

**Comment by liorn** [comment]
@ori-moisis note I changed another doc

File: README.md [hunk 1]
```diff
@@ -30,8 +30,9 @@ See [the vscode package documentation](packages/vscode/README.md#installation)
 
 ### Building from source
 
-  1. Install [yarn](https://yarnpkg.com/en/docs/install).
-  2. Fetch dependencies and build:
+  1. Install Node.js 12.22.7 . You can download it directly from [here](https://nodejs.org/en/download/releases/), or use [Node Version Manager (NVM)](https://github.com/nvm-sh/nvm) to install it.
```
Comment (ori-moisis) [inline README.md:33]: I think we don't need such a specific version, saying node 12 would be sufficient IMO

**Review by ori-moisis — APPROVED** [review]

**Pull request merged**